stream-compositions/services/legal-entity-composition-service/src/main/java/com/backbase/stream/compositions/legalentity/core/service/impl/LegalEntityPostIngestionServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+import com.backbase.stream.compositions.legalentity.core.mapper.LegalEntityMapper;
+import com.backbase.stream.compositions.legalentity.core.model.LegalEntityResponse;
+import com.backbase.stream.compositions.legalentity.core.service.LegalEntityPostIngestionService;
+import com.backbase.stream.compositions.product.client.ProductCompositionApi;
+import com.backbase.stream.compositions.product.client.model.ProductIngestionResponse;
+import com.backbase.stream.compositions.product.client.model.ProductPullIngestionRequest;
+import com.backbase.stream.legalentity.model.JobProfileUser;
+import com.backbase.stream.legalentity.model.LegalEntity;
+import com.backbase.stream.legalentity.model.User;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class LegalEntityPostIngestionServiceImpl implements LegalEntityPostIngestionService {
+
+    private final EventBus eventBus;
+
+    private final LegalEntityConfigurationProperties config;
+
+    private final ProductCompositionApi productCompositionApi;
+
+    private final LegalEntityMapper mapper;
+
+
+    @Override
+    public Mono<LegalEntityResponse> handleSuccess(LegalEntityResponse res) {
+        return Mono.just(res)
+                .doOnNext(r -> log.info("Legal entities ingestion completed successfully. {}",
+                        res.getLegalEntity().getInternalId()))
+                .flatMap(this::processChains)
+                .doOnNext(this::processSuccessEvent)
+                .doOnNext(r -> {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Ingested legal entity: {}", res.getLegalEntity());
+                    }
+                });
+    }
+
+    private Mono<LegalEntityResponse> processChains(LegalEntityResponse res) {
+        Mono<LegalEntityResponse> productChainMono;
+        boolean isProductChainEnabled = res.getProductChainEnabledFromRequest() == null ?
+                config.isProductChainEnabled():
+                res.getProductChainEnabledFromRequest();
+
+
+        if (!isProductChainEnabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("Product Chain is disabled");
+            }
+            productChainMono = Mono.just(res);
+        } else if (config.isProductChainAsync()) {
+            productChainMono = ingestProductsAsync(res);
+        } else {
+            productChainMono = ingestProducts(res);
+        }
+
+        return productChainMono;
+
+    }
+
+    private Mono<LegalEntityResponse> ingestProducts(LegalEntityResponse res) {
+        return buildProductPullRequest(res)
+                .flatMap(productCompositionApi::pullIngestProduct)
+                .onErrorResume(this::handleProductError)
+                .doOnSuccess(response -> {
+                    if (log.isDebugEnabled()) {
```

Comment:
this conditional checks are redundant when we are using slf4j with logback

Reply:
On second thought, I agree with you. This was over-engineering trying to save costs with the underlying operations (which aren't many to begin with) but ends up adding cost of checking debugEnabled every single time.

This'll be removed in the next PR

Reply:
Fixed